CSS Functions - Update descriptions (#21376)
mdn/content: files/en-us/web/css/css_functions/index.md

## Transform functions

These functions are used when the {{CSSxRef("&lt;transform-function&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) is used as a value of {{CSSxRef("transform")}} property.

### Translate functions

> - {{CSSxRef("transform-function/translateX", "translateX()")}}
>   - : Translate an element horizontally.
> - {{CSSxRef("transform-function/translateY", "translateY()")}}
>   - : Translate an element vertically.
> - {{CSSxRef("transform-function/translateZ", "translateZ()")}}
>   - : Translate an element along the z-axis.
> - {{CSSxRef("transform-function/translate", "translate()")}}
>   - : Translate an element on the 2D plane.
> - {{CSSxRef("transform-function/translate3d", "translate3d()")}}
>   - : Translate an element in 3D space.

Reviewer: ```suggestion
- {{CSSxRef("transform-function/translateX", "translateX()")}}
  - : Translates an element horizontally.
- {{CSSxRef("transform-function/translateY", "translateY()")}}
  - : Translates an element vertically.
- {{CSSxRef("transform-function/translateZ", "translateZ()")}}
  - : Translates an element along the z-axis.
- {{CSSxRef("transform-function/translate", "translate()")}}
  - : Translates an element on the 2D plane.
- {{CSSxRef("transform-function/translate3d", "translate3d()")}}
  - : Translates an element in 3D space.
```

### Rotation functions

- {{CSSxRef("transform-function/rotateX", "rotateX()")}}

>   - : Rotate an element around the horizontal axis.
> - {{CSSxRef("transform-function/rotateY", "rotateY()")}}
>   - : Rotate an element around the vertical axis.
> - {{CSSxRef("transform-function/rotateZ", "rotateZ()")}}
>   - : Rotate an element around the z-axis.
> - {{CSSxRef("transform-function/rotate", "rotate()")}}
>   - : Rotate an element around a fixed point on the 2D plane.
> - {{CSSxRef("transform-function/rotate3d", "rotate3d()")}}
>   - : Rotate an element around a fixed axis in 3D space.

Reviewer: ```suggestion
  - : Rotates an element around the horizontal axis.
- {{CSSxRef("transform-function/rotateY", "rotateY()")}}
  - : Rotates an element around the vertical axis.
- {{CSSxRef("transform-function/rotateZ", "rotateZ()")}}
  - : Rotates an element around the z-axis.
- {{CSSxRef("transform-function/rotate", "rotate()")}}
  - : Rotates an element around a fixed point on the 2D plane.
- {{CSSxRef("transform-function/rotate3d", "rotate3d()")}}
  - : Rotates an element around a fixed axis in 3D space.
```

### Scaling functions

- {{CSSxRef("transform-function/scaleX", "scaleX()")}}

>   - : Scale an element up or down horizontally.
> - {{CSSxRef("transform-function/scaleY", "scaleY()")}}
>   - : Scale an element up or down vertically.
> - {{CSSxRef("transform-function/scaleZ", "scaleZ()")}}
>   - : Scale an element up or down along the z-axis.
> - {{CSSxRef("transform-function/scale", "scale()")}}
>   - : Scale an element up or down on the 2D plane.
> - {{CSSxRef("transform-function/scale3d", "scale3d()")}}
>   - : Scale an element up or down in 3D space.

Reviewer: ```suggestion
  - : Scales an element up or down horizontally.
- {{CSSxRef("transform-function/scaleY", "scaleY()")}}
  - : Scales an element up or down vertically.
- {{CSSxRef("transform-function/scaleZ", "scaleZ()")}}
  - : Scales an element up or down along the z-axis.
- {{CSSxRef("transform-function/scale", "scale()")}}
  - : Scales an element up or down on the 2D plane.
- {{CSSxRef("transform-function/scale3d", "scale3d()")}}
  - : Scales an element up or down in 3D space.
```

### Skew functions

- {{CSSxRef("transform-function/skewX", "skewX()")}}

>   - : Skew an element in the horizontal direction.
> - {{CSSxRef("transform-function/skewY", "skewY()")}}
>   - : Skew an element in the vertical direction.
> - {{CSSxRef("transform-function/skew", "skew()")}}
>   - : Skew an element on the 2D plane.

Reviewer: ```suggestion
  - : Skews an element in the horizontal direction.
- {{CSSxRef("transform-function/skewY", "skewY()")}}
  - : Skews an element in the vertical direction.
- {{CSSxRef("transform-function/skew", "skew()")}}
  - : Skews an element on the 2D plane.
```

### Matrix functions

- {{CSSxRef("transform-function/matrix", "matrix()")}}

>   - : Describe a homogeneous 2D transformation matrix.
> - {{CSSxRef("transform-function/matrix3d", "matrix3d()")}}
>   - : Describe a 3D transformation as a 4×4 homogeneous matrix.

Reviewer: ```suggestion
  - : Describes a homogeneous 2D transformation matrix.
- {{CSSxRef("transform-function/matrix3d", "matrix3d()")}}
  - : Describes a 3D transformation as a 4×4 homogeneous matrix.
```

### Perspective functions

- {{CSSxRef("transform-function/perspective", "perspective()")}}

>   - : Set the distance between the user and the z=0 plane.

Reviewer: ```suggestion
  - : Sets the distance between the user and the z=0 plane.
```

## Math functions

The math functions allow CSS numeric values to be written as mathematical expressions.

### Basic arithmetic

- {{CSSxRef("calc")}}

>   - : Performe basic arithmetic calculations on numerical values.

Reviewer: ```suggestion
  - : Performs basic arithmetic calculations on numerical values.
```

### Comparison functions

- {{CSSxRef("min")}}

>   - : Calculate the smallest of a list of values.
> - {{CSSxRef("max")}}
>   - : Calculate the largest of a list of values.
> - {{CSSxRef("clamp")}}
>   - : Calculate the central of a minimum, central, and maximum values.

Reviewer: ```suggestion
  - : Calculates the smallest of a list of values.
- {{CSSxRef("max")}}
  - : Calculates the largest of a list of values.
- {{CSSxRef("clamp")}}
  - : Calculates the central of a minimum, central, and maximum values.
```

### Stepped value functions

- {{CSSxRef("round", "round()")}} {{Experimental_Inline}}

>   - : Calculate a rounded number based on a rounding strategy.
> - {{CSSxRef("mod", "mod()")}} {{Experimental_Inline}}
>   - : Calculate a modulus (with the same sign as the divisor) when dividing one number by another.
> - {{CSSxRef("rem", "rem()")}} {{Experimental_Inline}}
>   - : Calculate a remainder (with the same sign as the dividend) when dividing one number by another.

Reviewer: ```suggestion
  - : Calculates a rounded number based on a rounding strategy.
- {{CSSxRef("mod", "mod()")}} {{Experimental_Inline}}
  - : Calculates a modulus (with the same sign as the divisor) when dividing one number by another.
- {{CSSxRef("rem", "rem()")}} {{Experimental_Inline}}
  - : Calculates a remainder (with the same sign as the dividend) when dividing one number by another.
```

### Trigonometric functions

> - {{CSSxRef("sin")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric sine of a number.
> - {{CSSxRef("cos")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric cosine of a number.
> - {{CSSxRef("tan")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric tangent of a number.
> - {{CSSxRef("asin")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric inverse sine of a number.
> - {{CSSxRef("acos")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric inverse cosine of a number.
> - {{CSSxRef("atan")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric inverse tangent of a number.
> - {{CSSxRef("atan2")}} {{Experimental_Inline}}
>   - : Calculate a trigonometric inverse tangent of two-numbers in a plane.

Reviewer: ```suggestion
- {{CSSxRef("sin")}} {{Experimental_Inline}}
  - : Calculates the trigonometric sine of a number.
- {{CSSxRef("cos")}} {{Experimental_Inline}}
  - : Calculates the trigonometric cosine of a number.
- {{CSSxRef("tan")}} {{Experimental_Inline}}
  - : Calculates the trigonometric tangent of a number.
- {{CSSxRef("asin")}} {{Experimental_Inline}}
  - : Calculates the trigonometric inverse sine of a number.
- {{CSSxRef("acos")}} {{Experimental_Inline}}
  - : Calculates the trigonometric inverse cosine of a number.
- {{CSSxRef("atan")}} {{Experimental_Inline}}
  - : Calculates the trigonometric inverse tangent of a number.
- {{CSSxRef("atan2")}} {{Experimental_Inline}}
  - : Calculates the trigonometric inverse tangent of two-numbers in a plane.
```

### Exponential functions

- {{CSSxRef("pow")}} {{Experimental_Inline}}

>   - : Calculate a base raised to the power of a number.
> - {{CSSxRef("sqrt")}} {{Experimental_Inline}}
>   - : Calculate the square root of a number.
> - {{CSSxRef("hypot")}} {{Experimental_Inline}}
>   - : Calculate the square root of the sum of squares of its arguments.
> - {{CSSxRef("log")}} {{Experimental_Inline}}
>   - : Calculate the logarithm of a number.
> - {{CSSxRef("exp")}} {{Experimental_Inline}}
>   - : Calculate `e` raised to the power of a number.

Reviewer: ```suggestion
  - : Calculates the base raised to the power of a number.
- {{CSSxRef("sqrt")}} {{Experimental_Inline}}
  - : Calculates the square root of a number.
- {{CSSxRef("hypot")}} {{Experimental_Inline}}
  - : Calculates the square root of the sum of the squares of its arguments.
- {{CSSxRef("log")}} {{Experimental_Inline}}
  - : Calculates the logarithm of a number.
- {{CSSxRef("exp")}} {{Experimental_Inline}}
  - : Calculatex `e` raised to the power of a number.
```

### Sign-related functions

- {{CSSxRef("abs")}} {{Experimental_Inline}}

>   - : Calculate the absolute value of a number.
> - {{CSSxRef("sign")}} {{Experimental_Inline}}
>   - : Calculate the sign (positive or negative) of the number.

Reviewer: ```suggestion
  - : Calculates the absolute value of a number.
- {{CSSxRef("sign")}} {{Experimental_Inline}}
  - : Calculates the sign (positive or negative) of the number.
```

## Filter functions

The {{CSSxRef("&lt;filter-function&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) represents a graphical effect that can change the appearance of an input image. It is used in the {{CSSxRef("filter")}} and {{CSSxRef("backdrop-filter")}} properties.

- {{CSSxRef("filter-function/blur", "blur()")}}

>   - : Increase the image gaussian blur.
> - {{CSSxRef("filter-function/brightness", "brightness()")}}
>   - : Brighten or darken an image.
> - {{CSSxRef("filter-function/contrast", "contrast()")}}
>   - : Increase or decrease the image contrast.
> - {{CSSxRef("filter-function/drop-shadow", "drop-shadow()")}}
>   - : Apply a drop shadow behind an image.
> - {{CSSxRef("filter-function/grayscale", "grayscale()")}}
>   - : Convert an image to grayscale.
> - {{CSSxRef("filter-function/hue-rotate", "hue-rotate()")}}
>   - : Change the overall hue of an image.
> - {{CSSxRef("filter-function/invert", "invert()")}}
>   - : Invert the colors of an image.
> - {{CSSxRef("filter-function/opacity", "opacity()")}}
>   - : Add transparancy to an image.
> - {{CSSxRef("filter-function/saturate", "saturate()")}}
>   - : Change the overall saturation of an image.
> - {{CSSxRef("filter-function/sepia", "sepia()")}}
>   - : Increase the sepia of an image.

Reviewer: ```suggestion
  - : Increases the image gaussian blur.
- {{CSSxRef("filter-function/brightness", "brightness()")}}
  - : Brightens or darkens an image.
- {{CSSxRef("filter-function/contrast", "contrast()")}}
  - : Increases or decreases the image contrast.
- {{CSSxRef("filter-function/drop-shadow", "drop-shadow()")}}
  - : Applies a drop shadow behind an image.
- {{CSSxRef("filter-function/grayscale", "grayscale()")}}
  - : Converts an image to grayscale.
- {{CSSxRef("filter-function/hue-rotate", "hue-rotate()")}}
  - : Changes the overall hue of an image.
- {{CSSxRef("filter-function/invert", "invert()")}}
  - : Inverts the colors of an image.
- {{CSSxRef("filter-function/opacity", "opacity()")}}
  - : Adds transparency to an image.
- {{CSSxRef("filter-function/saturate", "saturate()")}}
  - : Changes the overall saturation of an image.
- {{CSSxRef("filter-function/sepia", "sepia()")}}
  - : Increases the sepia of an image.
```

## Color functions

The {{CSSxRef("color_value","&lt;color&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) specifies different color representations.

- {{CSSxRef("color_value/color", "color()")}} {{Experimental_Inline}}

>   - : Allows a color to be specified in a particular, specified colorspace (rather than the implicit sRGB colorspace that most of the other color functions operate in).

Reviewer: ```suggestion
  - : Allows a color to be set in the specified colorspace (instead of the implicit sRGB colorspace that most of the other color functions operate in).
```

- {{CSSxRef("color_value/color-mix", "color-mix()")}} {{Experimental_Inline}}
  - : Takes two {{CSSxRef("color_value","color")}} values and returns the result of mixing them in a given colorspace by a given amount.
- {{CSSxRef("color_value/color-contrast", "color-contrast()")}} {{Experimental_Inline}}
  - : Takes a {{CSSxRef("color_value","color")}} value and compares it to a list of other {{CSSxRef("color_value","color")}} values, selecting the one with the highest contrast from the list.
- {{CSSxRef("color_value/device-cmyk", "device-cmyk()")}} {{Experimental_Inline}}

>   - : Used to express CMYK colors in a device-independent way.

Reviewer: ```suggestion
  - : Defines CMYK colors in a device-independent way.
```

  - : HWB (short for Hue-Whiteness-Blackness) is another method of specifying colors, similar to HSL.
- {{CSSxRef("color_value/lab", "lab()")}} {{Experimental_Inline}}
  - : Lab color is device-independent and specifies physical measurements of color.
- {{CSSxRef("color_value/lch", "lch()")}} {{Experimental_Inline}}
  - : LCH color is device-independent and specifies color using polar coordinates for chroma and hue.
- {{CSSxRef("color_value/oklab", "oklab()")}} {{Experimental_Inline}}
  - : OKLab color is device-independent and specifies physical measurements of color.
- {{CSSxRef("color_value/oklch", "oklch()")}} {{Experimental_Inline}}
  - : OKLCH color is device-independent and specifies color using polar coordinates for chroma and hue.
- {{CSSxRef("color_value/rgb", "rgb()")}}
  - : The RGB color model defines a given color according to its red, green, and blue components. An optional alpha component represents the color's transparency.
- {{CSSxRef("color_value/rgba", "rgba()")}}
  - : The RGB color model defines a given color according to its red, green, and blue components. The alpha component represents the color's transparency.

## Image functions

> The {{CSSxRef("&lt;image&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) represents graphical representation of images or a gradients.

Reviewer: ```suggestion
The {{CSSxRef("&lt;image&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) provides graphical representation of images or gradients.
```

- {{CSSxRef("gradient/repeating-conic-gradient","repeating-conic-gradient()")}}
  - : Is similar to `conic-gradient()` and takes the same arguments, but it repeats the color stops infinitely in all directions so as to cover its entire container.

### Image functions

- {{CSSxRef("image/image","image()")}} {{Experimental_Inline}}
  - : Defines an {{CSSxRef("&lt;image&gt;")}} in a similar fashion to the {{CSSxRef("url", "url()")}} function, but with added functionality including specifying the image's directionality and fallback images for when the preferred image is not supported.
- {{CSSxRef("image/image-set","image-set()")}} {{Experimental_Inline}}
  - : A method of letting the browser pick the most appropriate CSS image from a given set, primarily for high pixel density screens.
- {{CSSxRef("cross-fade")}}
  - : Can be used to blend two or more images at a defined transparency.
- {{CSSxRef("element")}}
  - : Defines an {{CSSxRef("&lt;image&gt;")}} value generated from an arbitrary HTML element.
- {{CSSxRef("paint")}}
  - : Defines an {{CSSxRef("&lt;image&gt;")}} value generated with a PaintWorklet.

## Counter functions

> CSS counter functions are generally used with the {{CSSxRef("content")}} property, although in theory may be used wherever a {{CSSxRef("&lt;string&gt;")}} is supported.

Reviewer: ```suggestion
CSS counter functions are generally used with the {{CSSxRef("content")}} property, although in theory, they may be used wherever a {{CSSxRef("&lt;string&gt;")}} is supported.
```

- {{CSSxRef("counter")}}

>   - : Return a string representing the current value of the named counter, if there is one.
> - {{CSSxRef("counters")}}
>   - : Enable nested counters, returning a concatenated string representing the current values of the named counters, if there are any.
> - {{CSSxRef("symbols")}}
>   - : Define the counter styles inline, directly as the value of a property.

Reviewer: ```suggestion
  - : Returns a string representing the current value of the named counter if there is one.
- {{CSSxRef("counters")}}
  - : Enables nested counters, returning a concatenated string representing the current values of the named counters, if there are any.
- {{CSSxRef("symbols")}}
  - : Defines the counter styles inline, directly as the value of a property.
```

## Font functions

CSS font functions are used with the {{CSSxRef("font-variant-alternates")}} property to control the use of alternate glyphs.

> - {{CSSxRef("font-variant-alternates/stylistic", "stylistic()")}}
>   - : Enable stylistic alternates for individual characters. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `salt`, like `salt 2`.
> - {{CSSxRef("font-variant-alternates/styleset", "styleset()")}}
>   - : Enable stylistic alternatives for sets of characters. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `ssXY`, like `ss02`.
> - {{CSSxRef("font-variant-alternates/character-variant", "character-variant()")}}
>   - : Enable specific stylistic alternatives for characters. It is similar to `styleset()`, but doesn't create coherent glyphs for a set of characters; individual characters will have independent and not necessarily coherent styles. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `cvXY`, like `cv02`.
> - {{CSSxRef("font-variant-alternates/swash", "swash()")}}
>   - : Enable [swash](https://en.wikipedia.org/wiki/Swash_%28typography%29) glyphs. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType values `swsh` and `cswh`, like `swsh 2` and `cswh 2`.
> - {{CSSxRef("font-variant-alternates/ornaments", "ornaments()")}}
>   - : Enable ornaments, like [fleurons](https://en.wikipedia.org/wiki/Fleuron_%28typography%29) and other dingbat glyphs. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `ornm`, like `ornm 2`.
> - {{CSSxRef("font-variant-alternates/annotation", "annotation()")}}
>   - : Enable annotations, like circled digits or inverted characters. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `nalt`, like `nalt 2`.

Reviewer: ```suggestion
- {{CSSxRef("font-variant-alternates/stylistic", "stylistic()")}}
  - : Enables stylistic alternates for individual characters. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `salt`, like `salt 2`.
- {{CSSxRef("font-variant-alternates/styleset", "styleset()")}}
  - : Enables stylistic alternatives for sets of characters. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `ssXY`, such as `ss02`.
- {{CSSxRef("font-variant-alternates/character-variant", "character-variant()")}}
  - : Enables specific stylistic alternatives for characters. It is similar to `styleset()`, but doesn't create coherent glyphs for a set of characters; individual characters will have independent and not necessarily coherent styles. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `cvXY`, such as `cv02`.
- {{CSSxRef("font-variant-alternates/swash", "swash()")}}
  - : Enables [swash](https://en.wikipedia.org/wiki/Swash_%28typography%29) glyphs. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType values `swsh` and `cswh`, such as `swsh 2` and `cswh 2`.
- {{CSSxRef("font-variant-alternates/ornaments", "ornaments()")}}
  - : Enables ornaments such as [fleurons](https://en.wikipedia.org/wiki/Fleuron_%28typography%29) and other dingbat glyphs. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `ornm`, such as `ornm 2`.
- {{CSSxRef("font-variant-alternates/annotation", "annotation()")}}
  - : Enables annotations such as circled digits or inverted characters. The parameter is a font-specific name mapped to a number. It corresponds to the OpenType value `nalt`, such as `nalt 2`.
```

## Shape functions

> The {{CSSxRef("&lt;basic-shape&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) represents graphical shape. It is used in the {{CSSxRef("clip-path")}}, {{CSSxRef("offset-path")}}, and {{CSSxRef("shape-outside")}} properties.

Reviewer: ```suggestion
The {{CSSxRef("&lt;basic-shape&gt;")}} CSS [data type](/en-US/docs/Web/CSS/CSS_Types) represents a graphical shape. It is used in the {{CSSxRef("clip-path")}}, {{CSSxRef("offset-path")}}, and {{CSSxRef("shape-outside")}} properties.
```

- {{CSSxRef("basic-shape/circle","circle()")}}

>   - : Define a circle shape.
> - {{CSSxRef("basic-shape/ellipse","ellipse()")}}
>   - : Define an ellipse shape.
> - {{CSSxRef("basic-shape/inset","inset()")}}
>   - : Define an inset rectangle shape.
> - {{CSSxRef("basic-shape/polygon","polygon()")}}
>   - : Define a polygon shape.
> - {{CSSxRef("path")}}
>   - : Accept an SVG path string to enable a shape to be drawn.

Reviewer: ```suggestion
  - : Defines a circle shape.
- {{CSSxRef("basic-shape/ellipse","ellipse()")}}
  - : Defines an ellipse shape.
- {{CSSxRef("basic-shape/inset","inset()")}}
  - : Defines an inset rectangle shape.
- {{CSSxRef("basic-shape/polygon","polygon()")}}
  - : Defines a polygon shape.
- {{CSSxRef("path")}}
  - : Accepts an SVG path string to enable a shape to be drawn.
```

## Reference functions

The following functions are used as a value of properties to reference a value defined elsewhere.

- {{CSSxRef("attr")}}

>   - : Use attributes defined on HTML element.
> - {{CSSxRef("env")}}
>   - : Use user-agent defined as environment variable.
> - {{CSSxRef("url")}}
>   - : Use a file from a specific URL.
> - {{CSSxRef("var")}}
>   - : Use custom property value instead of any part of a value of another property.

Reviewer: ```suggestion
  - : Uses the attributes defined on HTML element.
- {{CSSxRef("env")}}
  - : Uses the user-agent defined as environment variable.
- {{CSSxRef("url")}}
  - : Uses a file from the specified URL.
- {{CSSxRef("var")}}
  - : Uses the custom property value instead of any part of a value of another property.
```

## Grid functions

The following functions are used to define a [CSS Grid](/en-US/docs/Web/CSS/CSS_Grid_Layout).

- {{CSSxRef("fit-content", "fit-content()")}}

>   - : Clamp a given size to an available size according to the formula `min(maximum size, max(minimum size, argument))`.
> - {{CSSxRef("minmax")}}
>   - : Define a size range greater-than or equal-to _min_ and less-than or equal-to _max_.
> - {{CSSxRef("repeat")}}
>   - : Represent a repeated fragment of the track list, allowing a large number of columns or rows that exhibit a recurring pattern.

Reviewer: ```suggestion
  - : Clamps a given size to an available size according to the formula `min(maximum size, max(minimum size, argument))`.
- {{CSSxRef("minmax")}}
  - : Defines a size range greater-than or equal-to _min_ and less-than or equal-to _max_.
- {{CSSxRef("repeat")}}
  - : Represents a repeated fragment of the track list, allowing a large number of columns or rows that exhibit a recurring pattern.
```
